unlock-js/src/unlockProvider.js

```diff
@@ -33,33 +33,21 @@ export default class UnlockProvider {
     return true
   }
 
-  async eth_accounts(respond) {
+  async eth_accounts() {
     // Must always return an array of addresses
     if (this.wallet) {
-      respond([this.wallet.address])
-    } else {
-      respond([])
+      return [this.wallet.address]
     }
+    return []
   }
 
-  send(args, cb) {
-    const { id, jsonrpc, method } = args
-    // Calling respond with some argument will call the callback with
-    // a "fake" JSON-RPC response constructed by the provider.
-    const respond = result => {
-      cb(null, {
-        id,
-        jsonrpc,
-        result,
-      })
-    }
-
-    try {
-      return this[method](respond)
-    } catch (err) {
+  async send(method, params) {
+    if (typeof this[method] === 'undefined') {
```

Comment:
@cellog it looks like methods on ES6 classes don't get defined in a way that `hasOwnProperty` works with, but this should be more or less equivalent and avoid the overhead of a `try/catch`